docs/src/pages/de/migration/0.21.0.md

```diff
@@ -0,0 +1,227 @@
+.env.local          # wird in allen Fällen geladen, ignoriert von git
+.env.[mode]         # wird nur im spezifizierten Modus geladen
+.env.[mode].local   # wird nur im spezifizierten Modus geladen, ignoriert von git
+```
+
+Aus Sicherheitsgründen sind ausschließlich Variablen mit dem Prefix `PUBLIC_` für deinen Code verfügbar. 
+
+```ini
+GEHEIMES_KENNWORT=kennwort123
+PUBLIC_IRGENDWER=dieser
+```
+
+In diesem Beispiel ist `PUBLIC_IRGENDWER` verfügbar als ìmport.meta.env.PUBLIC_IRGENDWER für Server- oder Client-seitigen Code, `GEHEIMES_KENNWORT` hingegen nicht.
+
+> In früheren Versionen wurde diesen Variablen das Prefix `SNOWPACK_PUBLIC` vorangestellt, und sie erforderten darüber hinaus das Plugin `@snowpack/plugin-env`.
+
+## Dateiendungen
+
+In Astro v21.0 muss auf Dateien mit Nennung ihrer tatsächlichen Erweiterung verwiesen werden, exakt so wie sie gespeichert sind. 
+
+```tsx
+// Div.tsx
+export default function Div(props) {
+  return <div />;
+}
+```
+
+In diesem Beispiel müsste auf `Div.tsx` mit `Div.tsx` verwiesen werden, nicht mit `Div.jsx`.
+
+```diff
+- import Div from './Div.jsx' // Astro v0.20
++ import Div from './Div.tsx' // Astro v0.21
+```
+
+Diese Änderung betrifft auch Style-Dateien.
+
+```scss
+// Div.scss
+div {
+  all: unset;
+}
+```
+
+```diff
+- <link rel="stylesheet" href={Astro.resolve('./Div.css')}>
++ <link rel="stylesheet" href={Astro.resolve('./Div.scss')}>
+```
+
+## Plugins
+
+In Astro v0.21 können Vite-Plugins innerhalb von `astro.config.mjs` konfiguriert werden.
+
+```js
+import { imagetools } from 'vite-imagetools';
+
+export default {
+  vite: {
+    plugins: [imagetools()],
+  },
+};
+```
+
+Um mehr über Vite-Plugins zu lernen, lies bitte Vites [Plugin Guide](https://vitejs.dev/guide/using-plugins.html).
+
+## Benutzerdefinierte Renderer
+
+In Astro v0.21 sollten Plugins ab jetzt `viteConfig()` verwenden.
+
+```diff
+// renderer-svelte/index.js
++ import { svelte } from '@sveltejs/vite-plugin-svelte';
+
+export default {
+  name: '@astrojs/renderer-svelte',
+  client: './client.js',
+  server: './server.js',
+-  snowpackPlugin: '@snowpack/plugin-svelte',
+-  snowpackPluginOptions: { compilerOptions: { hydratable: true } },
++  viteConfig() {
++    return {
++      optimizeDeps: {
++        include: ['@astrojs/renderer-svelte/client.js', 'svelte', 'svelte/internal'],
++        exclude: ['@astrojs/renderer-svelte/server.js'],
++      },
++      plugins: [
++        svelte({
++          emitCss: true,
++          compilerOptions: { hydratable: true },
++        }),
++      ],
++    };
++  },
+}
+```
+
+Um mehr über Vite-Plugins zu lernen, lies bitte Vites [Plugin Guide](https://vitejs.dev/guide/using-plugins.html). 
+
+> In früheren Versionen wurden diese Plugins mittels `snowpackPlugin` oder `snowpackPluginOptions` konfiguriert.
+
+## Änderungen bei der Gestaltung
+
+### Autoprefixer
+
+Autoprefixer wird nicht mehr automatisch eingebunden. Aktiviere Autoprefixer in zwei Schritten:
```

Comment:
While not in the English docs, It should be mentioned that vite has a default Postcss config that gets changed here.